pkg/utils/build/source_image.go

```diff
@@ -0,0 +1,108 @@
+package build
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/redhat-appstudio/e2e-tests/pkg/utils"
+	"github.com/tektoncd/pipeline/pkg/apis/pipeline/v1beta1"
+)
+
+const (
+	extraSourceSubDir = "extra_src_dir"
+	rpmSubDir         = "rpm_dir"
+	sourceTarFile     = "extra-src-0.tar"
+	shaValueRegex     = "[a-f0-9]{40}"
+	tarGzFileRegex    = ".tar.gz"
+)
+
+func GetBinaryImage(pr *v1beta1.PipelineRun) string {
+	for _, p := range pr.Spec.Params {
+		if p.Name == "output-image" {
+			return p.Value.StringVal
+		}
+	}
+	return ""
+}
+
+func SourceBuildEnabled(pr *v1beta1.PipelineRun) bool {
+	for _, p := range pr.Status.PipelineRunStatusFields.PipelineSpec.Params {
+		if p.Name == "build-source-image" {
+			if p.Default.StringVal == "true" {
+				return true
+			}
+		}
+	}
+	return false
+}
+
+func IsSourceFilesExistsInSourceImage(srcImage, gitUrl string) (bool, error) {
+	//Extract the src image locally
+	tmpDir, err := ExtractImage(srcImage)
+	defer os.RemoveAll(tmpDir)
+	if err != nil {
+		return false, err
+	}
+
+	// Check atleast one file present under extra_src_dir
+	absExtraSourceDirPath := filepath.Join(tmpDir, extraSourceSubDir)
+	fileNames, err := utils.GetFileNamesFromDir(absExtraSourceDirPath)
+	if err != nil {
+		return false, fmt.Errorf("error while getting files: %v", err)
+	}
+	if len(fileNames) == 0 {
+		fmt.Println("No files present under extra_src_dir")
+		return false, nil
+	}
+
+	//Check source tar exists
+	exists, err := utils.CheckFileExistsInDir(absExtraSourceDirPath, sourceTarFile)
+	if err != nil {
+		return false, fmt.Errorf("failed to check if file exists: %v", err)
+	}
+	if !exists {
+		fmt.Println("extra-src-0.tar is NOT found under extra_src_dir")
+		return false, nil
+	}
+
+	//Untar the source tar file
+	absSrcTarFilePath := filepath.Join(absExtraSourceDirPath, sourceTarFile)
+	err = utils.Untar(absExtraSourceDirPath, absSrcTarFilePath)
+	if err != nil {
+		return false, fmt.Errorf("error while untaring %s: %v", sourceTarFile, err)
+	}
+
+	//Again get the file list under extra_src_dir
+	fileNames, err = utils.GetFileNamesFromDir(absExtraSourceDirPath)
+	if err != nil {
+		return false, fmt.Errorf("error while getting files: %v", err)
+	}
+
+	//Get the <repo-name>-<git-sha>.tar.gz file name
+	repoName := utils.GetRepoName(gitUrl)
+	filePatternToFind := repoName + "-" + shaValueRegex + tarGzFileRegex
+	resultFiles := utils.FilterSliceUsingPattern(filePatternToFind, fileNames)
+	sourceGzTarFileName := resultFiles[0]
```

Comment:
`extra-src-0.tar` can be assumed to be the app source archive, but I don't recommend to make this assumption. What it includes depends on the implementation detail of BuildSourceImage.

However, if `extra_src_dir` directory has a single file `extra-src-0.tar` only, it is the app source archive.

Reply:
Changed it search for component source inside all the tar files now. Please take a look again.